verifier/src/main/kotlin/tech/figure/classification/asset/verifier/util/eventstream/providers/BlockApiEventStreamProvider.kt

```diff
@@ -46,14 +46,26 @@ class BlockApiEventStreamProvider(
             while (coroutineScope.isActive) {
                 (from..current).forEach { blockHeight ->
                     if (from > current) return@forEach
-                    process(blockHeight, onBlock, onEvent, onError, onCompletion)
+                    process(blockHeight, onBlock, onEvent, onError)
                     lastProcessed.set(blockHeight)
                 }
 
                 // Once we've met the current block, no need to keep spinning. Wait here for 4 seconds and process again.
                 delay(DEFAULT_BLOCK_DELAY_MS.milliseconds)
                 from = lastProcessed.incrementAndGet()
-                current = currentHeight()
+
+                runCatching {
+                    retry?.tryAction {
+                        current = currentHeight()
+                    } ?: run {
+                        current = currentHeight()
+                    }
+                }
+                    .onFailure { error ->
+                        onError(error)
+                    }
+
+                onCompletion(null)
```

Comment:
Moving this down out of `process()` such that we're not calling it after every block is retrieved. This better fits the description of the event (StreamCompleted): "When the event stream completes processing blocks (when it reaches the current block height), the flow will be
     * marked as completed and this function will execute. "